src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs

```diff
@@ -103,13 +104,34 @@ static async Task AwaitHandleCall(HttpContextServerCallContext serverCallContext
         /// This should only be called from client streaming calls
         /// </summary>
         /// <param name="httpContext"></param>
-        protected void DisableMinRequestBodyDataRate(HttpContext httpContext)
+        protected void DisableMinRequestBodyDataRateAndMaxRequestBodySize(HttpContext httpContext)
         {
             var minRequestBodyDataRateFeature = httpContext.Features.Get<IHttpMinRequestBodyDataRateFeature>();
             if (minRequestBodyDataRateFeature != null)
             {
                 minRequestBodyDataRateFeature.MinDataRate = null;
             }
+
+            var maxRequestBodySizeFeature = httpContext.Features.Get<IHttpMaxRequestBodySizeFeature>();
+            if (maxRequestBodySizeFeature != null)
+            {
+                if (!maxRequestBodySizeFeature.IsReadOnly)
+                {
+                    maxRequestBodySizeFeature.MaxRequestBodySize = null;
+                }
+                else
+                {
+                    httpContext.Features.Set<IHttpMaxRequestBodySizeFeature>(UnlimitedHttpMaxRequestBodySizeFeature.Instance);
```

Comment:
You don't need to do this.

Reply:
Why not? Won't MaxRequestBodySize throw an error is IsReadOnly is true.

Reply:
I re-read the IsReadOnly docs. Does it mean that is too late in the request for changing the body size to have any effect?

Reply:
> Does it mean that is too late in the request for changing the body size to have any effect?

Right.